applications/vanilla/controllers/api/ThemesApiController.php

```diff
@@ -0,0 +1,394 @@
+        ],
+        "scripts" => [
+            "type" => "json",
+            "file" => "scripts.json"
+        ],
+        "styles" => [
+            "type" => "css",
+            "file" => "styles.css"
+        ],
+        "inlineJs" => [
+            "type" => "js",
+            "file" => "inlineJs.js"
+        ]
+
+    ];
+
+    /* @var AddonManager */
+    private $addonManager;
+
+    /* @var ThemeModel */
+    private $themeModel;
+
+    /* @var ThemeAssetModel */
+    private $themeAssetModel;
+
+    /**
+     * @inheritdoc
+     */
+    public function __construct(
+        AddonManager $addonManager,
+        ThemeModel $themeModel,
+        ThemeAssetModel $themeAssetModel
+    ) {
+        $this->addonManager = $addonManager;
+        $this->themeModel = $themeModel;
+        $this->themeAssetModel = $themeAssetModel;
+    }
+
+    /**
+     * Get a theme assets.
+     *
+     * @param string $themeKey The unique theme key or theme ID.
+     * @return array
+     */
+    public function get(string $themeKey): array {
+
+        $in = $this->themeKeySchema('in')->setDescription('Get theme assets.');
+        $out = $this->themeResultSchema('out');
+
+        $themeAssets = $this->getThemeAssets($themeKey);
+        $themeAssets = $out->validate($themeAssets);
+        return $themeAssets;
+    }
+
+    /**
+     * Create new theme from parent theme.
+     *
+     * @param array $body Array of incoming params.
+     * @return array
+     */
+    public function post(array $body): array {
+        $this->permission("Garden.Settings.Manage");
+
+        $in = $this->themePostSchema('in')->setDescription('Post theme assets.');
+        $out = $this->themeResultSchema('out');
+        $body = $in->validate($body);
+        $themeID = $this->themeModel->insert($body);
+        if (isset($body['assets'])) {
+            foreach ($body['assets'] as $assetKey => $assetProps) {
+                if ($assetProps['type'] === 'html') {
+                    $assetRecord = [
+                        'themeID' => $themeID,
+                        'assetKey' =>$assetKey,
+                        'data' => $assetProps['body']
+                    ];
+                    $this->themeAssetModel->insert($assetRecord);
+                }
+            }
+        }
+
+        $theme = $this->getThemeAssets($themeID);
+        $themeAssets = $out->validate($theme);
+        return $themeAssets;
+    }
+
+    /**
+     * Get theme asset.
+     *
+     * @param string $id The unique theme key or theme ID (ex: keystone).
+     * @param string $assetKey Unique asset key (ex: header, footer, fonts, styles)
+     *        Note: assetKey can be filename (ex: header.html, styles.css)
+     *              in that case file content returned instaed of json structure
+     * @link https://github.com/vanilla/roadmap/blob/master/theming/theming-data.md#api
+     *
+     * @return array|Data
+     */
+    public function get_assets(string $id, string $assetKey) {
+
```

Comment:
This still needs a permission call This still needs a permission call:
https://docs.vanillaforums.com/developer/apiv2/api-endpoints/#check-permissions